mantidproject/mantid
# Q binning for non zero gamma angle

**Description of work.**
This PR changes the behavior of the q binning algorithm in SANSILLIntegration when the detector is not in front of the beam (gamma angle different from 0).
After some tries, binning from q_min to q_max with step (q_max - q_min)/pixels_by_row was determined a reasonable calculation (the original way of computing the binning cannot be extended over the whole range of the instrument).



**To test:**
`SANSILLAutoProcessTest` has been updated to take this change into account. Alternatively, one can use the files used for testing D16 to check the results' coherence.


Fixes #28425 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
